src/pip/_internal/req/req_file.py

```diff
@@ -481,6 +478,7 @@ def join_lines(lines_enum):
                 line = ' ' + line
             if new_line:
                 new_line.append(line)
+                assert primary_line_number is not None
```

Comment:
We can get rid of these asserts if we assign `primary_line_number = 0` as I see there is a certainty that `primary_line_number` will be assigned a value before we hit this or the other asserts, so putting the initial value as an integer instead of None would work, and I felt 0 as a more standard value for it.

Reply:
While that would silence mypy, it is less expressive because the algorithm must never yield 0 as a line number and the assert makes that explicit.